src/automatic_dantzig_wolfe.jl

```diff
@@ -0,0 +1,380 @@
+                               constraints_to_variables
+                            )
+    for k in keys(model.obj_dict)  # Check all names in the model
+        reference = model.obj_dict[k]
+        index_sets = _get_constraint_axes(reference)
+        if eltype(reference) <: JuMP.ConstraintRef # Add constraint
+            for c in reference
+                _add_constraint!(model_description, c, model, index_sets)
+            end
+        elseif eltype(reference) <: JuMP.VariableRef # Add variable
+            for v in reference
+                push!(variables, JuMP.index(v))
+            end
+        end
+    end
+    model_description.variables = variables
+    _add_anonymous_var_con!(model_description, model)
+    return model_description
+end
+
+# Adds a constraint to the ModelDescription object o
+function _add_constraint!(
+    o::ModelDescription,
+    c::JuMP.ConstraintRef,
+    model::JuMP.Model,
+    index_sets
+)
+    push!(o.constraints, c)
+    for r in index_sets
+        push!(o.axes, r)
+    end
+    o.constraints_to_axes[c] = index_sets
+    o.constraints_to_variables[c] = _get_variables_in_constraint(model, c)
+end
+
+# Computes the index sets of the constraint reference
+function _get_constraint_axes(constraint_ref::AbstractArray)
+    axs = axes(constraint_ref)
+    ds = JC.DenseAxisArray(constraint_ref, axs...)
+    return _get_constraint_axes(ds)
+end
+
+function _get_constraint_axes(constraint_ref::JC.DenseAxisArray)
+    axes_of_constraint = Array{BlockDecomposition.Axis,1}()
+    for a in constraint_ref.axes
+        if a != 1   # Axes of the form 1:1 do not matter (single constraints)
+            push!(axes_of_constraint, Axis(a))
+        end
+    end
+    return axes_of_constraint
+end
+
+function _get_constraint_axes(constraint_ref::JC.SparseAxisArray)
+    indices = eachindex(constraint_ref)
+    axes = Array{Set{Any}}(undef, 1)
+    for index in indices
+        for j in 1:length(index)
+            if length(axes) < length(index)
+                resize!(axes, length(index))
+            end
+            if !isassigned(axes, j)
+                axes[j] = Set()
+            end
+            push!(axes[j], index[j])
+        end
+    end
+    result = Array{BlockDecomposition.Axis,1}()
+    for a in axes
+        push!(result, Axis(a))
+    end
+    return result
+end
+
+function _get_variables_in_constraint(model::JuMP.Model, constraint::JuMP.ConstraintRef)
+    f = MOI.get(model, MathOptInterface.ConstraintFunction(), constraint)
+    variables = Set{MOI.VariableIndex}()
+    for term in f.terms
+        push!(variables, term.variable_index)
+    end
+    return variables
+end
+
+# Computes for the given axes two decomposition structures:
+# In the first one (bs0) all constraints *not indexed by any* axis from
+# axes are in the master, in the second one (bs1), all constraints *indexed
+# by at least one* axis from axes are in the master
+function get_block_structure(
+    axes::Array{<:Axis,1},
```

Comment:
```suggestion
    axes::Vector{Axis},
```
Why do you need `<:` ?

Reply:
Yes, julia documentation explains this : https://docs.julialang.org/en/v1/manual/types/#man-parametric-composite-types

> Concrete Point types with different values of T are never subtypes of each other

However, 

```julia
julia> using BlockDecomposition

julia> BlockDecomposition.Axis{Symbol(""), Int64} <: BlockDecomposition.Axis
true

julia> axes = BlockDecomposition.Axis[]
BlockDecomposition.Axis[]

julia> axis = BlockDecomposition.Axis(:boo, [1,2,3])
BlockDecomposition.Axis{:boo, Int64}(:boo, BlockDecomposition.AxisId{:boo, Int64}[1, 2, 3])

julia> push!(axes, axis)
1-element Vector{BlockDecomposition.Axis}:
 BlockDecomposition.Axis{:boo, Int64}(:boo, BlockDecomposition.AxisId{:boo, Int64}[1, 2, 3])
```